File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_ABC152_E.go

package main

import (
	"fmt"
	"math/big"
)

func main() {
	var n int
	fmt.Scan(&n)
	a := make([]int64, n)
	for i := range a {
		fmt.Scan(&a[i])
	}
	mod := int64(1e9 + 7)
	lcm := big.NewInt(1)
	for _, v := range a {
		lcm.Mul(lcm, big.NewInt(v))
		lcm.Div(lcm, big.NewInt(gcd(lcm.Int64(), v)))
	}
	modLcm := big.NewInt(0).Mod(lcm, big.NewInt(mod))
	var ans int64
	for _, v := range a {
		ans = (ans + modDiv(modLcm.Int64(), v, mod)) % mod
	}
	fmt.Println(ans)
}

func gcd(a, b int64) int64 {
	if b == 0 {
		return a
	}
	return gcd(b, a%b)
}

func modDiv(a, b, mod int64) int64 {
	return (a % mod) * modInv(b, mod) % mod
}

func modInv(x, mod int64) int64 {
	return modPow(x, mod-2, mod)
}

func modPow(b, e, mod int64) int64 {
	ans := int64(1)
	x := b % mod
	for y := e; y != 0; y >>= 1 {
		if (y & 1) == 1 {
			ans = (ans * x) % mod
		}
		x = (x * x) % mod
	}
	return ans
}

// //End of Code

File: LLM-Translations/GPT-4/avatar/Python/C/atcoder_ABC152_E.c
#include <stdio.h>
#define MOD 1000000007

long long gcd(long long a, long long b) {
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

long long power(long long a, long long b) {
    long long result = 1;
    while (b > 0) {
        if (b & 1)
            result = (result * a) % MOD;
        a = (a * a) % MOD;
        b = b >> 1;
    }
    return result;
}

int main() {
    int N;
    scanf("%d", &N);
    long long A[N];
    for (int i = 0; i < N; i++)
        scanf("%lld", &A[i]);
    long long _lcm = 1;
    for (int i = 0; i < N; i++)
        _lcm = (_lcm * A[i]) / gcd(_lcm, A[i]);
    long long ans = 0;
    for (int i = 0; i < N; i++)
        ans = (ans + power(A[i], MOD - 2)) % MOD;
    ans = (ans * _lcm) % MOD;
    printf("%lld\n", ans);
    return 0;
}

// End of Code
